llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -836,6 +836,22 @@ def IREELinalgExt_SetEncodingOp : IREELinalgExt_PureOp<"set_encoding",[
   }];
 }
 
+def IREELinalgExt_EncodingPaddingSizeOp: IREELinalgExt_PureOp<"encoding_padding_size", [Pure
+  ]> {
+  let summary = "xxx";
+  let description = [{
+    XXX
+  }];
+
+  let arguments = (ins AnyRankedTensor:$source,
+                       EncodingAttr:$encoding);
+  let results = (outs Variadic<Index>:$results);
+
+  let assemblyFormat = [{
+    attr-dict $source `:` type($source) $encoding `->` type($results)
```

Comment:
having `tensor<> <>` reads weird - you can make this `qualified($encoding)` so it prints out the attr name and makes more sense. it's useful to add some joining word/preposition between to indicate to readers what the attribute represents, as well, such as `in` or `as` ("the **tensor of this size/element type** is **encoded as blah** returning **a set of dimension values**")

Reply:
thanks, i didn't know about `qualified`.